test/e2e_node/services/kubelet.go

```diff
@@ -224,6 +253,94 @@ func (e *E2EServices) startKubelet() (*server, error) {
 	return server, server.start()
 }
 
+// splitKubeletConfigArgs parses args onto a KubeletConfiguration object and also returns the unknown args
+func splitKubeletConfigArgs(args []string) (*kubeletconfig.KubeletConfiguration, []string, error) {
+	kc, err := options.NewKubeletConfiguration()
+	if err != nil {
+		return nil, nil, err
+	}
+	fs := pflag.NewFlagSet("kubeletconfig", pflag.ContinueOnError)
+	options.AddKubeletConfigFlags(fs, kc)
+	known, other := splitKnownArgs(fs, args)
+	if err := fs.Parse(known); err != nil {
+		return nil, nil, err
+	}
+	return kc, other, nil
+}
+
+// splitKnownArgs splits argument list into those known by the flagset, and those not known
+// only tests for longhand args, e.g. prefixed with `--`
+// TODO(mtaufen): I don't think the kubelet has any shorthand args, but if it does we will need to modify this.
+func splitKnownArgs(fs *pflag.FlagSet, args []string) ([]string, []string) {
+	known := []string{}
+	other := []string{}
+	lastFlag := len(args)
+	for i := len(args) - 1; i >= 0; i-- {
+		if strings.HasPrefix(args[i], "--") {
+			if fs.Lookup(strings.TrimPrefix(args[i], "--")) == nil {
+				// flag is unknown, add flag and params to other
+				// prepend to maintain order
+				other = append(append([]string(nil), args[i:lastFlag]...), other...)
+				// cut from known
+			} else {
+				// flag is known, add flag and params to known
+				// prepend to maintain order
+				known = append(append([]string(nil), args[i:lastFlag]...), known...)
+			}
+			// mark the last location where we saw a flag
+			lastFlag = i
+		}
+	}
+	return known, other
+}
+
+// writeKubeletConfigFile writes the kubelet config file based on the args and returns the filename
+func writeKubeletConfigFile(internal *kubeletconfig.KubeletConfiguration) (string, error) {
+	// extract the KubeletConfiguration and convert to versioned
+	versioned := &v1alpha1.KubeletConfiguration{}
+	scheme, _, err := scheme.NewSchemeAndCodecs()
+	if err != nil {
+		return "", err
+	}
```

Comment:
Need to make any relative file path references either absolute or relative to the target directory for the config file